File: Python3/0434_number_of_segments_in_a_string.py
# https://leetcode.cn/problems/number-of-segments-in-a-string/

class Solution:
    def countSegments(self, s):
        """
        Inputs:
            s: str
        Outputs:
            res: int
        """
        res, n = 0, len(s)
        if n == 0:
            return res

        pre = s[0]
        for i in range(1, n):
            if s[i] == " " and pre != " ":
                res += 1
            pre = s[i]
        if s[n-1] != " ":
            res += 1
        return res

File: Cpp/0434_number_of_segments_in_a_string.cpp
// https://leetcode.cn/problems/number-of-segments-in-a-string/

class Solution {
public:
    int countSegments(string s) {
        int res = 0, n = s.size();
        if (n == 0) {
            return res;
        }
        char pre = s[0];
        for (int i = 0; i < n; ++i) {
            if ((s[i] == ' ') && (pre != ' ')) {
                res += 1;
            }
            pre = s[i];
        }
        if (s[n-1] != ' ') {
            res += 1;
        }
        return res;
    }
};
